src/docs/user/ProActiveUserGuide.adoc

```diff
@@ -1763,6 +1710,234 @@ println outFile.text
+```groovy
+jobid = variables.get("PA_JOB_ID")
+synchronizationapi.deleteChannel(jobid )
+println "Channel " + jobid + " deleted."
+```
+
+Here is output of the Key/Value workflow:
+```
+[28t2@trydev.activeeon.com;14:24:22] Channel 28 created.
+[28t0@trydev.activeeon.com;14:24:28] Sleeping 5 seconds
+[28t0@trydev.activeeon.com;14:24:33] Unlocking Task B
+[28t1@trydev.activeeon.com;14:24:40] Task B has been unlocked by Task A
+[28t3@trydev.activeeon.com;14:24:46] Channel 28 deleted.
+```
+
+NOTE: It is also possible to debug the operations performed on the Synchronization Store. This is possible by using the *Server Logs* associated with the job. Server Logs are very verbose, the _[Synchronization]_ pattern helps to find logs associated with the Synchronization API:
+```
+[2018-04-24 14:24:22,685 de72716883 INFO            o.o.p.s.u.TaskLogger] task 28t2 (Init) [Synchronization]Created new memory channel '28'
+[2018-04-24 14:24:22,786 de72716883 INFO            o.o.p.s.u.TaskLogger] task 28t2 (Init) [Synchronization][Channel=28] Put true on key 'lock', previous value was null
+[2018-04-24 14:24:34,431 de72716883 INFO            o.o.p.s.u.TaskLogger] task 28t0 (TaskA) [Synchronization][Channel=28] Put false on key 'lock', previous value was true
+[2018-04-24 14:24:46,882 de72716883 INFO            o.o.p.s.u.TaskLogger] task 28t3 (Clean) [Synchronization]Deleted memory channel '28'
+```
+
+The Complete Workflow example can be downloaded link:examples/KeyValue.xml[here].
+
+==== Example 2 : Simple Key/Value with Explicit Wait
+
+In this second example, we will show how one task can delay the execution of another task using the Synchronization API and an explicit wait call.
+
+This example is very similar to the first example, but instead of delaying the execution of TaskB using a selection script, TaskB will start its execution and explicitly call the Synchronization API to wait.
+
+ * Tasks *Init*, *TaskA* and *Clean* do not change
+
+ * *TaskB_Wait* is modified. The selection script is removed and instead the following code is used in the main task definition:
+```groovy
+println "Waiting for Task A"
+synchronizationapi.waitUntil(variables.get("PA_JOB_ID"), "lock", "{k, x -> x == false}")
+println "Task B has been unlocked by Task A"
+```
+
+As you can see, the https://www.activeeon.com/public_content/documentation/javadoc/dev/org/ow2/proactive/scheduler/synchronization/Synchronization.html#waitUntil-java.lang.String-java.lang.String-java.lang.String-[waitUntil] method expects a https://docs.oracle.com/javase/8/docs/api/java/util/function/BiPredicate.html?is-external=true[BiPredicate] defined as a string http://groovy-lang.org/closures.html[Groovy Closure].
+
+This predicate is evaluated on the *lock* binding, and waits until this binding *== false*.
```

Comment:
For the users not used to lambdas, it might be worth to explain the lambda in more depth, e.g. what is `k`, what is `x`.

Reply:
ok